### ISSUE TITLE
How to make it responsive?

### ISSUE THREAD

**Issue by nabjoern** [janpaepke/ScrollMagic#423]
I have a website that I have animated with GSAP and ScrollMagic.
Now I want to make it responsive. For this tweens need to be updated.
Do I have to rewrite the tweens, or is there a function to do it?

**Comment by trevordunn** [comment]
Yes you'll have to rewrite the tweens. What about them are you changing? Their duration or css vars?

However, scene durations can be made responsive to the viewport height simply by specifying a percentage (ex. "100%"). See the [Responsive Duration](http://scrollmagic.io/examples/basic/responsive_duration.html) example.

**Comment by valeriosillari** [comment]
i am working also on a similar problem.

at resize of window i want to change the duration. i cannot use the responsive option (percentage 100%) because one of the section is bigger than the viewport, so I need to init again scroll magic and set a new duration value in pixels.
any one has the same problem?

**Comment by b4z81** [comment]
take a look at this site: http://design-voilaah.com/mnd/app/ and to the related js:
http://design-voilaah.com/mnd/app/scripts/main.js

they managed the duration and the height of the scenes using dynamic values:

>  // calculate height and duration of each scene
        typicalHeight = parseInt( (newHeight - scene1Height) / 10, 10);
        curveHeight = typicalHeight/1.34;
        flatHeight = typicalHeight/0.86;

        $('.trigger', $cachedTriggers).each( function(i) {
            if (i > 0) { // for even i, we are in the curve, so the height is shorter
                (i % 2) ? $(this).height(curveHeight) : $(this).height(flatHeight);
            }
        });

> new ScrollMagic.Scene({triggerElement: ".trigger.scene2", duration: curveHeight * ratio})

**Comment by tim-basic** [comment]
I'm working on a similar problem too. I've been using scrollmagic to change the color of a hamburger menu depending on where it is on which section it is sitting on but on my durations go to hell when the orientation is changed or the length of something on a page increases after the initial load :/

**Comment by janpaepke** [comment]
Over ten years — sorry for the delay.

Making scroll-driven animations responsive was always tricky in v2 because tweens were defined with fixed values. The typical approach was to destroy and recreate scenes on resize. v3 approaches this differently — it tracks element positions via IntersectionObserver and ResizeObserver, so it automatically adapts when elements change size. Your animation library handles the actual motion, and you can use the `progress` event (0 to 1) to drive animations in a resolution-independent way.

---
**ScrollMagic v3 is now available** — a complete rewrite with an entirely new architecture and API. v2 will not receive further updates (2.0.9 is the final release).

👉 `npm install scrollmagic@next` — [Docs](https://github.com/janpaepke/ScrollMagic/blob/main/README.md) · [Release notes](https://github.com/janpaepke/ScrollMagic/releases/tag/v3.0.0-beta.0) · [npm](https://www.npmjs.com/package/scrollmagic/v/3.0.0-beta.0)
